physical/cassandra.go

```diff
@@ -0,0 +1,186 @@
+package physical
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	log "github.com/mgutz/logxi/v1"
+
+	"github.com/armon/go-metrics"
+	"github.com/gocql/gocql"
+)
+
+// CassandraBackend is a physical backend that stores data in Cassandra.
+type CassandraBackend struct {
+	sess  *gocql.Session
+	table string
+
+	logger log.Logger
+}
+
+// newCassandraBackend constructs a Cassandra backend using a pre-existing
+// keyspace and table.
+func newCassandraBackend(conf map[string]string, logger log.Logger) (Backend, error) {
+	splitArray := func(v string) []string {
+		return strings.FieldsFunc(v, func(r rune) bool {
+			return r == ','
+		})
+	}
+
+	var (
+		hosts       = splitArray(conf["hosts"])
+		keyspace    = conf["keyspace"]
+		table       = conf["table"]
+		consistency = gocql.LocalQuorum
+	)
```

Comment:
You might want to allow configuration of:

* username required The username to use for superuser access.
* password required The password corresponding to the given username.
* tls optional Whether to use TLS when connecting to Cassandra.
* insecure_tls optional Whether to skip verification of the server certificate when using TLS.
* pem_bundle optional Concatenated PEM blocks containing a certificate and private key; a certificate, private key, and issuing CA certificate; or just a CA certificate.
* pem_json optional JSON containing a certificate and private key; a certificate, private key, and issuing CA certificate; or just a CA certificate. For convenience format is the same as the output of the issue command from the pki backend; see the pki documentation.
* protocol_version optional The CQL protocol version to use. Defaults to 2.
* connect_timeout optional The connection timeout to use. Defaults to 5 seconds.

(see Cassandra Secret Backend).